lib/ui/src/settings/index.tsx

```diff
@@ -1,84 +1,94 @@
-import React, { FunctionComponent, SyntheticEvent } from 'react';
-import { Tabs, IconButton, Icons } from '@storybook/components';
-import { useStorybookApi } from '@storybook/api';
+import React, { FunctionComponent, SyntheticEvent, Fragment } from 'react';
+import { IconButton, Icons, FlexBar, TabBar, TabButton, ScrollArea } from '@storybook/components';
+import { useStorybookApi, API, useStorybookState } from '@storybook/api';
 import { Location, Route } from '@storybook/router';
 import { styled } from '@storybook/theming';
+import { GlobalHotKeys } from 'react-hotkeys';
 import { AboutPage } from './about_page';
 import { ReleaseNotesPage } from './release_notes_page';
 import { ShortcutsPage } from './shortcuts_page';
 
-const ABOUT = 'about';
-const SHORTCUTS = 'shortcuts';
-const RELEASE_NOTES = 'release-notes';
-
-export const Wrapper = styled.div`
-  div[role='tabpanel'] {
-    height: 100%;
-  }
-`;
-
-interface PureSettingsPagesProps {
-  activeTab: string;
-  changeTab: (tab: string) => {};
-  onClose: () => {};
-}
-
-const PureSettingsPages: FunctionComponent<PureSettingsPagesProps> = ({
-  activeTab,
-  changeTab,
-  onClose,
-}) => (
-  <Wrapper>
-    <Tabs
-      absolute
-      selected={activeTab}
-      actions={{ onSelect: changeTab }}
-      tools={
-        <IconButton
-          onClick={(e: SyntheticEvent) => {
+const TabBarButton: FunctionComponent<{ id: string; title: string }> = ({ id, title }) => (
+  <Location>
+    {({ navigate, path }) => {
+      const active = path.includes(`settings/${id}`);
+      return (
+        <TabButton
+          id={`tabbutton-${id}`}
+          className={['tabbutton'].concat(active ? ['tabbutton-active'] : []).join(' ')}
```

Comment:
Seems like we should be using the `selected` prop here? This feels brittle.

Reply:
We don't style this, the community wanted more theming options, and we wanted our theming system to stay relatively simple. this is the compromise we made